request/request.go

```diff
@@ -0,0 +1,148 @@
+package request
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+
+	"github.com/amp-labs/cli/logger"
+)
+
+type RequestClient struct {
+	Client *http.Client
+}
+
+// Header is a key/value pair that can be added to a request.
+type Header struct {
+	Key   string
+	Value string
+}
+
+func NewRequestClient() *RequestClient {
+	return &RequestClient{Client: http.DefaultClient}
+}
+
+// Put makes a PUT request to the desired URL, and unmarshalls the
+// response body into `result`.
+func (c *RequestClient) Put(ctx context.Context,
+	url string, reqBody any, result any, headers ...Header,
+) (*http.Response, error) {
+	req, err := makeJSONPutRequest(ctx, url, headers, reqBody)
+	if err != nil {
+		return nil, err
+	}
+
+	return c.makeRequestAndParseResult(req, result)
+}
+
+// Post makes a POST request to the desired URL, and unmarshalls the
+// response body into `result`.
+func (c *RequestClient) Post(ctx context.Context,
+	url string, reqBody any, result any, headers ...Header,
+) (*http.Response, error) {
+	req, err := makeJSONPostRequest(ctx, url, headers, reqBody)
+	if err != nil {
+		return nil, err
+	}
+
+	return c.makeRequestAndParseResult(req, result)
+}
+
+var ErrNone200Status = errors.New("error response from API")
+
+func (c *RequestClient) makeRequestAndParseResult(req *http.Request, result any) (*http.Response, error) {
+	logger.Debugf(">>> API REQUEST:\n%+v\n", req)
```

Comment:
The [httputil.DumpRequest](https://pkg.go.dev/net/http/httputil#DumpRequest) function might be useful here